src/twisted/runner/procmon.py

```diff
@@ -145,11 +175,16 @@ class ProcessMonitor(service.Service):
     @ivar _reactor: A provider of L{IReactorProcess} and L{IReactorTime}
         which will be used to spawn processes and register delayed calls.
 
+    @type log: C{Logger}
+    @ivar log: An instance of twisted.logger.Logger used for propagating
+        log messages from spawned processes.
+
     """
     threshold = 1
     killTime = 5
     minRestartDelay = 1
     maxRestartDelay = 3600
+    log = Logger()
```

Comment:
I was wondering if this should be `_log` to prevent making it a part of the public API as we would do in other cases, but I think this is correct for two reasons:

1. It's called by more than inside this class, since the public `LineLogger` calls it, and so making it private would require users who wish to use `LineLogger` (for whatever reason) have to implement a private API. If LineLogger were a private API, then this should be `_log`.

2. Having the logger defined here is important so that `ProcessMonitor` is given as the "source", not `LineLogger` (which is less useful). Therefore, we can't sidestep this problem by giving `LineLogger` their own loggers. This could be done by manually providing the source when loggers in `LineLogger`, but that feels fragile.

So, I think `log` here is the correct decision, even though it adds more public API surface. (Although, we should create an interface for Logger, so that we can say "an implementer of `ILogger`" in case we want to change the concrete implementation of Logger to something else while maintaining the same API.)

Reply:
I interpreted this as saying that creating the ILogger is something to be done in another PR, but let me know if that wasn't the expectation here.